engine.go

```diff
@@ -830,7 +852,14 @@ func (e *Engine) executeEvent(ctx *sql.Context, dbName, createEventStatement, us
 		return err
 	}
 
-	iter = finalizeIters(ctx, definitionNode, nil, iter)
+	iter, _, err = finalizeIters(ctx, definitionNode, nil, iter)
+	if err != nil {
+		clearAutocommitErr := clearAutocommitTransaction(ctx)
```

Comment:
should this error block be default behavior inside finalizeIters? also should the rest of the finalizers run or not if we hit the error?